web/src/search_suggestion.ts

```diff
@@ -6,26 +6,50 @@ import * as huddle_data from "./huddle_data";
 import * as narrow_state from "./narrow_state";
 import {page_params} from "./page_params";
 import * as people from "./people";
+import type {User} from "./people";
+import type {NarrowTerm} from "./state_data";
 import * as stream_data from "./stream_data";
 import * as stream_topic_history from "./stream_topic_history";
 import * as stream_topic_history_util from "./stream_topic_history_util";
 import * as typeahead_helper from "./typeahead_helper";
 
+type UserPillItem = {
+    id: number;
+    display_value: Handlebars.SafeString;
+    has_image: boolean;
+    img_src: string;
+    should_add_guest_user_indicator: boolean;
+};
+
+type InvalidItem = Omit<NarrowTerm, "operand"> & Partial<Pick<NarrowTerm, "operand">>;
```

Comment:
This is a very weird type name; why is it called this?

Reply:
I had a doubt on what should we call this ? InvalidTerm ?

Reply:
I think "invalid" is kinda a misleading name `TermPattern` might be most appropriate; they're not invalid globally, just categories of terms that are not possible given the terms we already have. But the property of being impossible/incompatible with existing terms isn't a property of the type; just a property of the set of things we're using the type for, so I don't think it belongs in the type name.